test/e2e/pods.go

```diff
@@ -34,11 +34,22 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+// createNamespaceIfDoesNotExist ensures that the namespace with specified name exists, or returns an error
+func createNamespaceIfDoesNotExist(c *client.Client, name string) (*api.Namespace, error) {
```

Comment:
`ensureNamespaceExists`? optimistically creating and tolerating an IsConflict() error would save you a request. Should probably tolerate a conflict error anyway